test/e2e_new/framework/azureidentity/azureidentity_helpers.go

```diff
@@ -67,14 +58,22 @@ func Create(input CreateInput) *aadpodv1.AzureIdentity {
 		// TODO: account for SP type
 		Spec: aadpodv1.AzureIdentitySpec{
 			Type:       input.IdentityType,
-			ResourceID: fmt.Sprintf(resourceIDTemplate, input.Config.SubscriptionID, input.Config.IdentityResourceGroup, input.IdentityName),
+			ResourceID: fmt.Sprintf(azure.ResourceIDTemplate, input.Config.SubscriptionID, input.Config.IdentityResourceGroup, input.IdentityName),
 			ClientID:   identityClientID,
 		},
 	}
 
-	Eventually(func() error {
-		return input.Creator.Create(context.TODO(), azureIdentity)
-	}, createTimeout, createPolling).Should(Succeed())
+	// For gatekeeper test case
+	if input.InvalidResourceID {
+		azureIdentity.Spec.ResourceID = fmt.Sprintf(invalidResourceIDTemplate, input.Config.SubscriptionID, input.IdentityName)
+		Eventually(func() error {
+			return input.Creator.Create(context.TODO(), azureIdentity)
```

Comment:
Not a blocker for this PR, but we should define contexts with timeout/deadline to ensure the tests don't hang until the pipeline timeout.

Reply:
Sounds good